libs/dh/shared/ui-paginator/src/lib/dh-shared-ui-paginator.component.ts

```diff
@@ -0,0 +1,99 @@
+/**
+ * @license
+ * Copyright 2020 Energinet DataHub A/S
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License2");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { Component, OnDestroy, inject, ViewChild, Input } from '@angular/core';
+import {
+  MatPaginator,
+  MatPaginatorIntl,
+  MatPaginatorModule,
+} from '@angular/material/paginator';
+import { TranslocoService } from '@ngneat/transloco';
+import { Subject, takeUntil } from 'rxjs';
+
+@Component({
+  standalone: true,
+  selector: 'dh-shared-ui-paginator',
+  imports: [MatPaginatorModule],
+  template: `
+    <mat-paginator
+      [length]="length"
+      [pageSize]="pageSize"
+      [pageSizeOptions]="pageSizeOptions"
+      [showFirstLastButtons]="true"
+      [attr.aria-label]="ariaLabel"
+    >
+    </mat-paginator>
+  `,
+})
+export class DhSharedUiPaginatorComponent implements OnDestroy {
+  @ViewChild(MatPaginator, { static: true }) private paginator!: MatPaginator;
+
+  @Input() length?: number;
+  @Input() pageSizeOptions: number[] = [50, 100, 150, 200, 250];
+  @Input() pageSize?: number;
+
+  ariaLabel?: string;
+
+  private matPaginatorIntl = inject(MatPaginatorIntl);
+  private translocoService = inject(TranslocoService);
+  private destroy$ = new Subject<void>();
+
+  getPaginator = () => this.paginator;
```

Comment:
Suggestion: Have you considered making this a getter function instead? The function call makes it seem like it has a non-trivial cost (fx. I would be hesitant to call this in ngOnChanges, even though it is basically free). Although `this.paginator.paginator` looks wrong, perhaps `this.paginator.instance`?